libminifi/include/processors/MergeContent.h

```diff
@@ -125,6 +127,127 @@ class BinaryConcatenationMerge : public MergeBin {
 };
 
 
+// Archive Class
+class ArchiveMerge {
+public:
+  // Nest Callback Class for read stream
+  class ReadCallback: public InputStreamCallback {
+  public:
+    ReadCallback(uint64_t size, struct archive *arch, struct archive_entry *entry) :
+        buffer_size_(size), arch_(arch), entry_(entry) {
+    }
+    ~ReadCallback() {
+    }
+    int64_t process(std::shared_ptr<io::BaseStream> stream) {
+      uint8_t buffer[buffer_size_];
+      int64_t ret = 0;
+      uint64_t read_size;
+      ret = stream->read(buffer, buffer_size_);
+      if (!stream)
```

Comment:
Checking whether stream is null after you attempt a function call on it seems counterintuitive.